### PR TITLE
Display better error messages for http error codes

### LINKED ISSUE #68: Better error message on network dashboard the backend is not reachable
-- _**Conclusion after discussion with @abizjak: Better error message on network dashboard the backend is not reachable**_ --

**Bug Description**
Wrong display of transaction lookup for multisig transaction that is not finalized yet:
dashboard_multisig transfer

After few seconds, probably after transaction is finalized, the display is correct:
dashboard_multisig transfer_2

transaction hash: `009ee4b5680de1234c4c5a2f1d9ac59d487642df1e0eba2803842d59513fc212`

**Steps to Reproduce**
- made multisig simple transfer in desktop wallet
- copied&pasted transaction hash quickly in network dashboard > transaction lookup
- enter

**Expected Result**
Transaction not finalized or sth similar.

**Actual Result**
See above.

**Versions**
- Network dashboard 1.0.1
- DW 1.2.0
- Chrome

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Closes #68.

## Changes

- Display a more readable error instead of http body for http error codes.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Comment by limemloh** [comment]
image

File: src/client/elm/Widgets.elm [hunk 1]
```diff
@@ -126,17 +126,23 @@ errorToString error =
...
-            "Server gateway timeout, try again later"
+            else if 500 <= code && code < 600 then
+                case code of
+                    504 ->
+                        "Server gateway timeout, try again later"
 
-        Http.BadStatus 400 ->
-            "Verify your information and try again"
+                    502 ->
+                        "Server bad gateway, try again later"
```
Comment (abizjak) [inline src/client/elm/Widgets.elm:139]: I think this is too technical, I recommend a more understandable phrasing of something like "Server not reachable. Try again later."

Comment (limemloh) [reply src/client/elm/Widgets.elm]: Okay, I'll change this to be the error for any `5xx` code

File: src/elm/Api.elm [hunk 2]
```diff
@@ -154,7 +154,7 @@ blockInfoDecoder =
 
 getTransactionStatus : Config -> T.TxHash -> (ApiResult TransactionStatusResponse -> msg) -> Cmd msg
 getTransactionStatus cfg txHash msg =
-    getMiddleware cfg ("/v1/transactionStatus/" ++ txHash) transactionStatusResponseDecoder msg
+    getMiddleware { middlewareUrl = "" } ("/v1/transactionStatus/" ++ txHash) transactionStatusResponseDecoder msg
```
Comment (Bargsteen) [inline src/elm/Api.elm:157]: Why was this change made? As far as I can tell the default config also uses `""` when in production mode.
If this change is necessary, then the `Config` parameter is unused and should be removed 😊

**Review by Bargsteen — CHANGES_REQUESTED** [review]
Looks good, I just have one minor issue.

File: src/elm/Api.elm (hunk 2 above)
Comment (limemloh) [reply src/elm/Api.elm:157]: this should not have been changed, it was only for debugging.
Nice catch

**Review by Bargsteen — APPROVED** [review]

**Review by abizjak — APPROVED** [review]

**Pull request merged**